CMakeLists.txt

```diff
@@ -244,9 +244,16 @@ option(SWIFT_BUILD_CLANG_OVERLAYS
   "Build Swift overlays for the clang builtin modules"
   TRUE)
 
-option(SWIFT_BUILD_DYNAMIC_SDK_OVERLAY
-    "Build dynamic variants of the Swift SDK overlay"
-    TRUE)
+# The SDK overlay is provided by the SDK itself on Darwin platforms.
+if (SWIFT_HOST_VARIANT_SDK IN_LIST SWIFT_DARWIN_PLATFORMS)
+  option(SWIFT_BUILD_DYNAMIC_SDK_OVERLAY
+      "Build dynamic variants of the Swift SDK overlay"
+      FALSE)
+else()
+  option(SWIFT_BUILD_DYNAMIC_SDK_OVERLAY
+      "Build dynamic variants of the Swift SDK overlay"
+      TRUE)
+endif()
```

Comment:
We should do this in a separate PR, not least because this isn't actually sufficient to stop it from building the Darwin module; the issue is this in `stdlib/public`:
```
if(SWIFT_BUILD_SDK_OVERLAY OR SWIFT_BUILD_TEST_SUPPORT_MODULES)
  add_subdirectory(Platform)
endif()
```
As `stdlib/CMakeLists.txt` makes clear:
```
option(SWIFT_BUILD_TEST_SUPPORT_MODULES
       "Whether to build StdlibUnittest and other test support modules. Defaults to On when SWIFT_BUILD_SDK_OVERLAY is On, or when SWIFT_INCLUDE_TESTS is On."
       "${SWIFT_BUILD_TEST_SUPPORT_MODULES_default}")
```
So if `SWIFT_INCLUDE_TESTS` is enabled, which it is by default, then we still build the overlays.

Reply:
(There's also @kubamracek's `embedded-darwin` module to worry about…)